### ISSUE TITLE
Installing Wireshark dissector

### ISSUE THREAD

**Issue by adlacilla** [AFITWiSec/EZ-Wave#18]
Hi,

I was able to get through the installation process successfully up to this point. The other steps in the install Wireshark dissector were completed successfully up to the make step.
When running make, this is what happens:

lacilla@ubuntu:~/Documents/Projects/wireshark$ make
  PERL     version.h
version.h unchanged.
make  all-recursive
make[1]: Entering directory `/home/lacilla/Documents/Projects/wireshark'
Making all in tools
make[2]: Entering directory `/home/lacilla/Documents/Projects/wireshark/tools'
Making all in lemon
make[3]: Entering directory `/home/lacilla/Documents/Projects/wireshark/tools/lemon'
make[3]: Nothing to be done for `all'.
make[3]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/tools/lemon'
make[3]: Entering directory `/home/lacilla/Documents/Projects/wireshark/tools'
make[3]: Nothing to be done for `all-am'.
make[3]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/tools'
make[2]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/tools'
Making all in wsutil
make[2]: Entering directory `/home/lacilla/Documents/Projects/wireshark/wsutil'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/wsutil'
Making all in writecap
make[2]: Entering directory `/home/lacilla/Documents/Projects/wireshark/writecap'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/writecap'
Making all in wiretap
make[2]: Entering directory `/home/lacilla/Documents/Projects/wireshark/wiretap'
make  all-am
make[3]: Entering directory `/home/lacilla/Documents/Projects/wireshark/wiretap'
make[3]: Nothing to be done for `all-am'.
make[3]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/wiretap'
make[2]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/wiretap'
Making all in epan
make[2]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan'
make  all-recursive
make[3]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan'
Making all in compress
make[4]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan/compress'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan/compress'
Making all in crypt
make[4]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan/crypt'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan/crypt'
Making all in ftypes
make[4]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan/ftypes'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan/ftypes'
Making all in dfilter
make[4]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan/dfilter'
make  all-am
make[5]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan/dfilter'
make[5]: Nothing to be done for `all-am'.
make[5]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan/dfilter'
make[4]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan/dfilter'
Making all in dissectors
make[4]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan/dissectors'
make  all-recursive
make[5]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan/dissectors'
Making all in dcerpc
make[6]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan/dissectors/dcerpc'
make[6]: Nothing to be done for `all'.
make[6]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan/dissectors/dcerpc'
make[6]: Entering directory `/home/lacilla/Documents/Projects/wireshark/epan/dissectors'
  CC       packet-afit-encapse.lo
packet-afit-encapse.c: In function 'dissect_afit_encap':
packet-afit-encapse.c:52:3: error: implicit declaration of function 'tvb_new_subset' [-Werror=implicit-function-declaration]
   next_tvb = tvb_new_subset(tvb, offset, tvb_captured_length_remaining(tvb,offset), tvb_reported_length(tvb));
   ^
packet-afit-encapse.c:52:12: error: assignment makes pointer from integer without a cast [-Werror]
   next_tvb = tvb_new_subset(tvb, offset, tvb_captured_length_remaining(tvb,offset), tvb_reported_length(tvb));
            ^
packet-afit-encapse.c: In function 'proto_register_afit_encap':
packet-afit-encapse.c:103:49: error: enum conversion when passing argument 4 of 'register_dissector_table' is invalid in C++ [-Werror=c++-compat]
                                                 FT_UINT8, BASE_DEC); 
                                                 ^
In file included from packet-afit-encapse.c:2:0:
../../epan/packet.h:181:33: note: expected 'ftenum_t' but argument is of type 'int'
 WS_DLL_PUBLIC dissector_table_t register_dissector_table(const char *name,
                                 ^
packet-afit-encapse.c:103:49: error: too few arguments to function 'register_dissector_table'
                                                 FT_UINT8, BASE_DEC); 
                                                 ^
In file included from packet-afit-encapse.c:2:0:
../../epan/packet.h:181:33: note: declared here
 WS_DLL_PUBLIC dissector_table_t register_dissector_table(const char *name,
                                 ^
packet-afit-encapse.c: In function 'proto_reg_handoff_afit_encap':
packet-afit-encapse.c:114:2: error: passing argument 1 of 'create_dissector_handle' from incompatible pointer type [-Werror]
  afit_encap_handle = create_dissector_handle (dissect_afit_encap, proto_afit_encap);
  ^
In file included from packet-afit-encapse.c:2:0:
../../epan/packet.h:531:34: note: expected 'dissector_t' but argument is of type 'void (*)(struct tvbuff_t *, struct packet_info *, struct proto_tree *)'
 WS_DLL_PUBLIC dissector_handle_t create_dissector_handle(dissector_t dissector,
                                  ^
cc1: all warnings being treated as errors
make[6]: *** [packet-afit-encapse.lo] Error 1
make[6]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan/dissectors'
make[5]: *** [all-recursive] Error 1
make[5]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan/dissectors'
make[4]: *** [all] Error 2
make[4]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan/dissectors'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/home/lacilla/Documents/Projects/wireshark/epan'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/lacilla/Documents/Projects/wireshark'
make: *** [all] Error 2
lacilla@ubuntu:~/Documents/Projects/wireshark$ 

Is there something i'm missing here? Any help with dealing with these errors is appreciated.

**Comment by cwbadenhop** [comment]
We haven't migrated to the latest version of wireshark. Are you building
against the correct version (see docs)?

**Comment by adlacilla** [comment]
what docs are you referring to?

**Comment by adlacilla** [comment]
if you are referring to the README , I am currently using wireshark 1.12.1. We have followed the install instructions on the README to the letter and installing only what was necessary to complete each step.

**Comment by cwbadenhop** [comment]
Yes I was referencing the readme. Sorry for the crappy advice...

The reported errors in your OP are in regard to Wireshark library calls and
data structures. While that leads me to believe there is a missing include
file, path, or environment variable misconfiguration, it still doesn't
explain why everything else built fine up to that point (unless its an
autoconfig issue...but I'm thinking out loud here). I will have to try to
build with Wireshark 1.12.1 and see if I can replicate the issue. Perhaps
there is an implicit step in the instructions...

Sadly, at the moment I can only suggest rebuilding wireshark from scratch
(i.e., make clean) or trying to build against Wireshark 2.0.1, which is the
version I use. More to come.

**Comment by adlacilla** [comment]
Hey,

Just wanted to check in and see if any progress was made. I did attempt to install a newer version of wireshark but ran into the same issue. Any other ideas?

**Comment by cwbadenhop** [comment]
I had time to build wireshark with Z-Wave dissectors over the weekend. I was unable to reproduce the compile errors you listed for wireshark versions 1.12.1 and 2.0.1 (they both built without issue) and downloaded from https://www.wireshark.org/download/src/all-versions/.

Are you copying the EZ-Wave files in setup/wireshark to ./wireshark/epan/dissectors?

Could you post your config log file for wireshark, which is generated after you run ./configure in the wireshark directory?
